### DISCUSSION TITLE
PyPy Support

### DISCUSSION THREAD

**Discussion by cirospaciari** [emmett-framework/granian#21] [Q&A] [answered]
I saw the commit: https://github.com/emmett-framework/granian/commit/b884e867463a8b1b642a052a08c242a7c00b84e3
And I really happy to see PyPy being supported!
I never saw PyO3 working with PyPy before.

Py03 converts to Cython or Python C API? in PyPy CFFI is the way to go, but CFFI has a lot of overhead on CPython.
HPy will solve all this. I see this in Py03 issues: https://github.com/PyO3/pyo3/issues/1113

Hopping to see granian getting a performance boost by using PyPy!

**Comment by gi0baro** [comment, marked as answer]
AFAIK, PyO3 always relies on FFI, not Cython.

Unfortunately, at least from local tests, since `uvloop` is not compatible with PyPy, the performance is comparable with CPython. The support is intended to give chance to developers to use the Python interpreter they prefer, not for performance reasons.

**Reply by cirospaciari** [reply]
Granian uses raw asyncio right? In a few days, I will release a SelectorEventLoop for PyPy (which is slow for CPython but is an FFI alternative for PyPy) maybe it will boost performance in PyPy for granian. Can I fork granian to test it?

**Reply by gi0baro** [reply]
Sure thing!
